clang/lib/Lex/LiteralSupport.cpp

```diff
@@ -1032,6 +1033,21 @@ NumericLiteralParser::NumericLiteralParser(StringRef TokSpelling,
 
       isFloat = true;
       continue;  // Success.
+    // C++23 5.13.4 [lex.fcon]
+    case 'b':
+    case 'B':
+      if (!isFPConstant)
+        break; // Error for integer constant.
+      if (s + 3 < ThisTokEnd && (s[1] == 'f' || s[1] == 'F') && s[2] == '1' &&
+          s[3] == '6') {
```

Comment:
This allows for `Bf` and `bF` which are not supposed to be allowed to work.
I'd rather we use StringRef::compare